packages/webpack-rtl-plugin/src/index.js

```diff
@@ -37,7 +37,9 @@ class WebpackRTLPlugin {
 
 								// Compute the filename
 								const baseName = path.basename( asset, '.css' );
-								const filename = asset.replace( baseName, `${ baseName }.rtl` );
```

Comment:
This replaces the first instance of basename, so an asset path like `test/foo/dist/foo.css` would turn into `test/foo.rtl/dist/foo.css`. The new fix correctly changes that to `test/foo/dist/foo.rtl.css`.